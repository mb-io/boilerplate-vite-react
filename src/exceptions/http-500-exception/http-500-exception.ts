import { AxiosError } from 'axios'

import { HttpException } from '../http-exception/http-exception'

export class Http500Exception extends HttpException {
  constructor(message?: string, error?: AxiosError) {
    super(500, message ?? 'Internal Server Error', error)
  }
}
