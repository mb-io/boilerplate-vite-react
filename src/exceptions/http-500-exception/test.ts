import { describe, it, expect } from 'vitest'
import { Http500Exception } from './http-500-exception'
import { AxiosError } from 'axios'

describe('Http500Exception', () => {
  it('should create an exception with default message', () => {
    const exception = new Http500Exception()
    expect(exception.status).toBe(500)
    expect(exception.message).toBe('Internal Server Error')
    expect(exception).toMatchSnapshot()
  })

  it('should create an exception with custom message', () => {
    const message = 'Custom internal server error message'
    const exception = new Http500Exception(message)
    expect(exception.status).toBe(500)
    expect(exception.message).toBe(message)
    expect(exception).toMatchSnapshot()
  })

  it('should create an exception with custom message and AxiosError', () => {
    const message = 'Custom internal server error message'
    const axiosError = new AxiosError('Axios error message')
    const exception = new Http500Exception(message, axiosError)
    expect(exception.status).toBe(500)
    expect(exception.message).toBe(message)
    expect(exception.error).toBe(axiosError)
    expect(exception).toMatchSnapshot()
  })
})
