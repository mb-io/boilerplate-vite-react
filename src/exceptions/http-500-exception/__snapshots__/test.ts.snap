// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Http500Exception > should create an exception with custom message 1`] = `[Error: Custom internal server error message]`;

exports[`Http500Exception > should create an exception with custom message and AxiosError 1`] = `[Error: Custom internal server error message]`;

exports[`Http500Exception > should create an exception with default message 1`] = `[Error: Internal Server Error]`;
